paddle/platform/device_context.h

```diff
@@ -74,8 +75,7 @@ class CUDADeviceContext : public DeviceContext {
  public:
   explicit CUDADeviceContext(const GPUPlace gpu_place) : gpu_place_(gpu_place) {
     GPUPlaceGuard guard(gpu_place_);
-    paddle::platform::throw_on_error(cudaStreamCreate(&stream_),
-                                     "cudaStreamCreate failed");
+    PADDLE_ENFORCE(cudaStreamCreate(&stream_), "cudaStreamCreate failed");
```

Comment:
Given that 

```cpp
PADDLE_ENFORCE(cudaStreamCreate(&stream_))
```

is shorter than 

```cpp
PADDLE_ENFORCE(cudaStreamCreate(&stream_) == CUDA_SUCCESS);
```
and it is compatible with the use of the word "enforce" in English,  I support that we adopt @gangliao 's proposal -- to make `PADDLE_ENFORCE` handle the condition.